chains/ethereum/listener.go

```diff
@@ -169,9 +169,13 @@ func (l *listener) getDepositEventsForBlock(latestBlock *big.Int) error {
 	// read through the log events and handle their deposit event if handler is recognized
 	for _, log := range logs {
 		var m msg.Message
-		destId := msg.ChainId(log.Topics[1].Big().Uint64())
-		rId := msg.ResourceIdFromSlice(log.Topics[2].Bytes())
-		nonce := msg.Nonce(log.Topics[3].Big().Uint64())
+		parsedLog, err := l.bridgeContract.BridgeFilterer.ParseDeposit(log)
+		if err != nil {
+			return fmt.Errorf("failed to parse Deposit event: %w", err)
+		}
+		destId := msg.ChainId(parsedLog.DestinationChainID)
+		rId := msg.ResourceIdFromSlice(parsedLog.ResourceID[:])
```

Comment:
Is there a point to use this util function?
Seems msg.ResourceId(parsedLog.ResourceID) works the same.

Reply:
Fixed.